_includes/includes/ui/keyboard-details.php

```diff
@@ -90,16 +90,28 @@ public static function get_tier_from_request($tier = 'stable') {
       return $tier;
     }
 
-    protected function map_license($s) {
+    protected static function map_license($s) {
       $license_map = ['mit' => 'MIT'];
       return array_key_exists($s, $license_map) ? $license_map[$s] : $s;
     }
 
-    protected function download_box($url, $title, $description, $class, $linktitle, $platform, $mode='standalone') {
+    protected static function download_box($url, $title, $description, $class, $linktitle, $platform, $mode='standalone') {
       $urlbits = explode('/', $url);
       $filename = array_pop($urlbits);
       $id = self::$id;
-      $downloadlink = "<a class='download-link binary-download' href='download?id=$id&platform=$platform&mode=$mode' onclick='return downloadBinaryFile(this);'><span>$linktitle</span></a>";
+      global $embed;
+      
+      $e_filename = urlencode($filename);
+      $e_id = urlencode($id);
+
+      if($embed != 'none') {
+        // note: if embed != none, mode should was be standalone
+        $url = "keyman:download?filename=$e_filename&url=".urlencode("https://keyman.com/keyboard/download?id=$e_id&platform=$platform&mode=$mode");
```

Comment:
Looking down the road when the mobile apps will browser/download keyboards through these links, will the apps need to handle `keyman:download`?

Reply:
Possibly; we'll cross that design bridge when we come to the update.